src/realm/query_expression.hpp

```diff
@@ -3739,12 +3798,29 @@ class Maximum : public BaseAggregateOperation<T, Maximum<T>> {
     }
 };
 
+template <>
+class Maximum<Decimal128> : public BaseAggregateOperation<Decimal128, Maximum<Decimal128>> {
+public:
+    static Decimal128 initial_value()
+    {
+        return Decimal128("-inf");
+    }
+    static Decimal128 apply(Decimal128 a, Decimal128 b)
+    {
+        return std::max(a, b);
+    }
+    static std::string description()
+    {
+        return "@max";
+    }
+};
+
 template <typename T>
 class Sum : public BaseAggregateOperation<T, Sum<T>> {
 public:
     static T initial_value()
     {
-        return T();
+        return T(0);
```

Comment:
Rather than doing this we should make `Decimal128()` be the zero value, as it is for every other numeric type.